src/vos/evt_iter.c

```diff
@@ -385,6 +392,7 @@ evt_iter_probe(daos_handle_t ih, enum evt_iter_opc opc,
 		D_GOTO(out, rc = -DER_NO_HDL);
 
 	enta = &iter->it_entries;
+	ent_array_reset(tcx, enta);
```

Comment:
Should we call ent_array_reset() inside evt_ent_array_fill() before checking whether the tree is empty or not? Otherwise if there are (new) callers for evt_ent_array_fill(), they may also miss to reset the ent_array.

Reply:
Probably, if evt_ent_array_fill() will never allow an non-empty ent_array in the future, we may force array reset inside of evt_ent_array_fill().
I just followed current implementation which replies on caller to pass in a ent_array (empty or partial filled), so I moved the reset out of the evt_ent_array_fill().